OpenRA.Game/Widgets/WidgetUtils.cs

```diff
@@ -243,7 +243,8 @@ public static string ChooseInitialMap(string initialUid)
 						return false;
 
 					// Maps with bots disabled confuse new players
-					if (m.Map.Players.Any(s => !s.Value.AllowBots))
+					var players = new MapPlayers(m.Map.PlayerDefinitions);
+					if (players.Players.Any(s => !s.Value.AllowBots))
```

Comment:
You could save some performace by caching this "allows bots" (or maybe even "is ideal map") value on the MapPreview.